vignettes/examples/hierarchical_linear_general.Rmd

```diff
@@ -0,0 +1,31 @@
+### Hierarchical linear regression in general conditional formulation
+
+A hierarchical, Bayesian linear regression model using the iris data, with random intercepts and slopes for each of the three species.
+The slopes and intercepts for each species are *correlated* in this example. We allow every species to have a species specific slope for `Sepal.Length`.
+
+
+```{r hierarchical_linear_general_greta}
+int  <- normal(0, 10)
+coef <- normal(0, 10)
+sd   <- cauchy(0, 3, truncation = c(0, Inf))
+
+n.species  <- length(unique(iris$Species))
+species_id <- as.numeric(iris$Species)
+
+Z     <- matrix(0, nrow(iris), n.species * 2)
+gamma <- zeros(n.species * 2)
+
+for (s in unique(species_id)) {
+  Z[species_id == s, s * 2  - 1] <- 1
```

Comment:
I think this is a simpler (and probably easier for users) way of creating the same matrix:
```r
Z <- model.matrix(~ Species + Sepal.Length * Species - 1, data = iris)
```

It's in a slightly different order (intercepts first), but otherwise the same

Reply:
Thanks, I don't know why I didn't do it like this right from the start ...